### DISCUSSION TITLE
How to useMTproto Proxy in bot?

### DISCUSSION THREAD

**Discussion by DineshValor** [masterking32/MasterHamsterKombatBot#296] [Q&A] [answered]
![Screenshot_20240831-063645](https://github.com/user-attachments/assets/d6f20ee1-b6aa-4944-b394-060f82d5f3f2)

**Comment by Fy0urM** [comment]
not supported, if I'm not mistaken MTProto is a proxy protocol for telegram and you need to use telegram libraries
but it can't be used just for web requests, only for telegram client.

**Reply by DineshValor** [reply]
> not supported, if I'm not mistaken MTProto is a proxy protocol for telegram and you need to use telegram libraries but it can't be used just for web requests, only for telegram client.

I tried Socks5 proxy even not working for me in bot

**Reply by Fy0urM** [reply, marked as answer]
> > not supported, if I'm not mistaken MTProto is a proxy protocol for telegram and you need to use telegram libraries but it can't be used just for web requests, only for telegram client.
> 
> I tried Socks5 proxy even not working for me in bot

to use socks proxy you need to install additional library `pip install requests[socks]`

**Comment by Fy0urM** [comment]
read this https://github.com/masterking32/MasterHamsterKombatBot/discussions/133
